### PR TITLE
chore: Add GitHub workflow to run nox checks

### PR DESCRIPTION
**Pull request by luator**
Tests are run on all supported Python versions.  Linters only on 3.10 (which is default on Ubuntu 22.04 so currently probably to most common development version for us).

### CODE REVIEW

File: .github/workflows/nox.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Nox
+
+on: [pull_request]
```
Comment (mseitzer) [inline .github/workflows/nox.yml:3]: I would additionally run it on pushes (at least) to master. 

Can be done as follows:

```
on: 
  push:
    branches:
      - "master"
  pull_request:
```

**Review by mseitzer — APPROVED** [review]
Looks good, thanks!

The only thing I'm wondering if there is any caching that could be done to speed up creation of the Nox environments.

File: .github/workflows/nox.yml (hunk 1 above)
Comment (luator) [reply .github/workflows/nox.yml]: Added (and also workflow_dispatch, so we can run it manually if we want).
I'm wondering a bit if running on master is really needed, though.  Instead I'd rather set branch rules such that one can only merge via pull request if the branch is up-to-date and checks have passed.  In this case running again on master after the merge would just repeat the same checks.

**Comment by luator** [comment]
Regarding caching: It is in principle possible but saving/loading the cache into the instance that is running the workflow also takes time, so I don't think it would make a big difference.  That was at least my experience with caching containers vs pulling them again for each job.

File: .github/workflows/nox.yml (hunk 1 above)
Comment (mseitzer) [reply .github/workflows/nox.yml]: That sounds acceptable if merges to master really ever only happen that way. Also, does the Github UI render the CI checkmark correctly on the main page if it's only run on a branch?

Comment (luator) [reply .github/workflows/nox.yml]: I think it doesn't show any check mark on the main page.  If we want this, we would probably need to add a badge to the README.  In that case, it is probably really necessary to run on push to master.

**Pull request merged**